src/tplr/comms.py

```diff
@@ -108,6 +114,44 @@ def __init__(
         self.client_semaphore = asyncio.Semaphore(30)  # Limit concurrent connections
         self.retry_config = {"max_attempts": 3, "backoff_base": 1.5}
 
+    async def _get_s3_client(self, bucket: Bucket):
+        """
+        Returns a persistent s3_client for the given bucket credentials.
+        We create it if we haven't already, else reuse the existing client.
+        """
+        key = (bucket.access_key_id, bucket.secret_access_key, bucket.account_id)
+        if key in self._s3_clients:
+            return self._s3_clients[key]
+
+        # Create the base URL
+        endpoint_url = self.get_base_url(bucket.account_id)
+
+        # Create the client (equivalent to `async with`, but we store the client persistently)
+        new_client = self.session.create_client(
+            "s3",
+            endpoint_url=endpoint_url,
+            region_name=CF_REGION_NAME,
+            config=client_config,
```

Comment:
in the config, could you also add please `tcp_keepalive` ?  https://github.com/boto/botocore/blob/develop/botocore/config.py#L207C11-L207C24

Reply:
Added in [7848811](https://github.com/tplr-ai/templar/pull/181/commits/7848811adf0d669b934c37b2e9e01cf82f55f49c)